File: src/game/entity/Entity.java
package game.entity;

import game.math.Vector2;

public class Entity {

    Vector2 position;

    public Entity(Vector2 position) {
        this.position = position;
    }

    public void update(long delta) {

    }

    public float getX() {
        return position.getX();
    }

    public float getY() {
        return position.getY();
    }

    public void translateX(float x) {
        position.translateX(x);
    }

    public void translateY(float y) {
        position.translateY(y);
    }
}

File: src/game/entity/Entity.kt
package game.entity

import game.math.Vector2

open class Entity(internal var position: Vector2) {

    open fun update(delta: Long) {

    }

    val x: Float
        get() = position.x

    val y: Float
        get() = position.y

    fun translateX(x: Float) {
        position.translateX(x)
    }

    fun translateY(y: Float) {
        position.translateY(y)
    }
}
